crates/dslab-iaas/src/experiments.rs

```diff
@@ -0,0 +1,60 @@
+//! Tools for launching experiments with multiple environment configurations
+
+use std::cell::RefCell;
+use std::rc::Rc;
+
+use sugars::{rc, refcell};
+
+use dslab_core::Simulation;
+
+use crate::core::config::exp_config::ExperimentConfig;
+use crate::simulation::CloudSimulation;
+
+/// Callbacks on simulation events
+pub trait SimulationCallbacks {
+    fn on_simulation_start(&mut self, _sim: Rc<RefCell<CloudSimulation>>) {
+        // custom callback
+    }
+
+    fn on_step(&mut self, _sim: Rc<RefCell<CloudSimulation>>) {
```

Comment:
Надо добавить возвращаемое значение - если оно `false`, то останавливать симуляцию (позволит реализовать кастомные правила остановки):

```suggestion
    fn on_step(&mut self, _sim: Rc<RefCell<CloudSimulation>>) -> bool {
```